src/RequestHandler.cpp

```diff
@@ -86,13 +85,22 @@ void RequestHandler::showTopicPartitionOffsets(
   }
 }
 
-void RequestHandler::consumePartitions(
-    std::pair<std::string, bool> MessageAndEOF, int &EOFPartitionCounter,
-    FlatbuffersTranslator &FlatBuffers) {
-  if (!MessageAndEOF.first.empty() &&
-      MessageAndEOF.first != "HiddenSecretMessageFromLovingNeutron") {
-    FlatBuffers.getFileID(&MessageAndEOF.first);
+void RequestHandler::consumePartitions(KafkaMessageMetadataStruct &MessageData,
+                                       int &EOFPartitionCounter,
+                                       FlatbuffersTranslator &FlatBuffers) {
+  if (!MessageData.PayloadToReturn.empty() &&
+      MessageData.PayloadToReturn != "HiddenSecretMessageFromLovingNeutron") {
+    std::string JSONMessage = FlatBuffers.getFileID(MessageData);
```

Comment:
It is strange to see a method called `getFileID()` and find it returns a JSON message. This is probably a sign that the method should be renamed. It maybe also be that it should be refactored into multiple methods, if part of its functionality is to get the file ID but it also generates the JSON.